pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -288,6 +288,9 @@ func (c *Cluster) UpdateActiveMgrLabel(daemonNameToUpdate string, prevActiveMgr
 			pod.SetLabels(labels)
 			_, err = c.context.Clientset.CoreV1().Pods(c.clusterInfo.Namespace).Update(c.clusterInfo.Context, &pods.Items[i], metav1.UpdateOptions{})
 			if err != nil {
+				// In case of failure we report as 'active manager' the previous value, this way
+				// we force refreshing mgrRoleLabelName label next time this function is called
+				currActiveMgr = prevActiveMgr
```

Comment:
What if there were multiple pods found to update, and this fails on the first one? Then the second one will effectively set `prevActiveMgr` again? Or maybe it's fine since the next overall attempt would fix it. But maybe instead we need a separate var to keep track of what should be returned from the method?

Reply:
I added a comment to clarify why are we iterating over all the pods and what happens when an update fails. As of the current design in case of failure we will just rollback the currentActiveMgr and report the previous value to keep trying the label update on the next call.